sphinx/themes/basic/static/searchtools.js

```diff
@@ -627,15 +640,18 @@ const Search = {
 
       // ensure that none of the excluded terms is in the search result
       if (
-        [...excludedTerms].some(
-          (term) =>
-            terms[term] === file
-            || titleTerms[term] === file
-            || (terms[term] || []).includes(file)
-            || (titleTerms[term] || []).includes(file),
-        )
+        [...excludedTerms].some((excludedTerm) => {
+          // Both mappings will contain either a single integer or a list of integers.
+          // Converting them to lists makes the comparison more readable.
+          let excludedTermFiles = [].concat(terms[excludedTerm]);
+          let excludedTitleFiles = [].concat(titleTerms[excludedTerm]);
+          return (
+            excludedTermFiles.includes(file)
+            || excludedTitleFiles.includes(file)
+          );
+        })
```

Comment:
```suggestion
        [...excludedTerms].some(
          (term) =>
            terms[term] === file
            || titleTerms[term] === file
            || (terms[term] || []).includes(file)
            || (titleTerms[term] || []).includes(file),
        )
```

Do we need this set of `excludedTerms` filtering changes?  I would suggest not modifying these lines unless strictly necessary.  Tests continue to pass when I revert this.

I acknowledge that the `break` to `continue` fixup is important though.

Reply:
Ok, I'll need to add another test then. The last conditions raise an error in some cases. I'll add it soon.

Reply:
Hi @jayaddison,
I added the "should exclude results where the file index is above 0." test case which raises an TypeError with the old code. It's very specific for the old implementation. 
The error occurs every time the `terms[term]` statement is a positive integer and not equal to the `file` variable. I think this bug was introduced when the datastructures were changed from actual file paths/names to file ids. 
The strings usually evaluate to `false` in this condition, but integers above 0 evaluate to `true`.